src/HomieNode.cpp

```diff
@@ -9,18 +9,48 @@ PropertyInterface::PropertyInterface()
 : _property(nullptr) {
 }
 
-void PropertyInterface::settable(const PropertyInputHandler& inputHandler) {
+PropertyInterface& PropertyInterface::settable(const PropertyInputHandler& inputHandler) {
   _property->settable(inputHandler);
+  return *this;
+}
+
+PropertyInterface& PropertyInterface::setName(const char* name) {
+  _property->setName(name);
+  return *this;
+}
+
+PropertyInterface& PropertyInterface::setUnit(const char* unit) {
+  _property->setUnit(unit);
+  return *this;
+}
+
+PropertyInterface& PropertyInterface::setDatatype(const char* datatype) {
+  _property->setDatatype(datatype);
+  return *this;
+}
+
+PropertyInterface& PropertyInterface::setFormat(const char* format) {
+  _property->setFormat(format);
+  return *this;
+}
+
+PropertyInterface& PropertyInterface::retained(const bool retained) {
+  _property->retained(retained);
+  return *this;
 }
 
 PropertyInterface& PropertyInterface::setProperty(Property* property) {
   _property = property;
   return *this;
 }
 
-HomieNode::HomieNode(const char* id, const char* type, const NodeInputHandler& inputHandler)
+HomieNode::HomieNode(const char* id, const char* name, const char* type, bool range, uint16_t lower, uint16_t upper, const NodeInputHandler& inputHandler)
```

Comment:
In all your examples the "id" is the same as "type".
"type" is actually optional, at least it doesn't say anything about the node in the controller perspective. 
Could you maybe remove it from the constructor and make it a chainable setter instead?

Reply:
Make all the array specific parameters a dedicated setter as well (`setArray(uint16_t lower, uint16_t upper)`). That way if arrays change at some point, we don't need to change the constructor.

Reply:
> "type" is actually optional, at least it doesn't say anything about the node in the controller perspective.

Type is a required attribute for Nodes in Homie 3.0.1, see here: https://homieiot.github.io/spec-core-v3_0_1/#node-attributes. That's why I have added it to the class constructor. Perhaps we can remove it for Homie 4.0, but that discussion doesn't belong here :wink: 

> Make all the array specific parameters a dedicated setter as well (setArray(uint16_t lower, uint16_t upper)). That way if arrays change at some point, we don't need to change the constructor.

I didn't want to change that part of the code too much, because it was already implemented, just the minimum necessary changes. What do you think @marvinroger?